e2e/test/scenarios/visualizations/table.cy.spec.js

```diff
@@ -320,10 +322,17 @@ describe("scenarios > visualizations > table", () => {
 });
 
 describe("scenarios > visualizations > table > conditional formatting", () => {
+  before(() => {
+    restore("default");
+    cy.signInAsAdmin();
+
+    setupWritableDB("postgres");
+    addPostgresDatabase("Writable Postgres12", true);
+  });
+
```

Comment:
In theory, this part can live inside `beforeEach`. It doesn't make a difference in an isolated test. I just wanted to enforce this pattern because it will be useful for `describe` blocks with multiple tests. In such cases, we can simply set up things once (in a `before` hook`), take a db snapshot and then restore for every single test (in `beforeEach`).

Single tests don't even require snapshot/restore.

Reply:
I think this is fine for a single test, but the reason why I put this in setup is that it is a potentially heavy operation that could take a few seconds - if we snapshot it, we never have to do it again, but if we do it across many tests, we're potentially wasting a bunch of time.

Reply:
The time we will "waste" is insignificant compared to a failure rate of more than 50%.
Which means you have to wait for about half an hour for tests that didn't fail to finish only to be able to re-run failed specs.

Compare that to 5 added seconds to a single spec.
This is exactly the reason I'm using `before` and not `beforeEach` hook. We will still do it only once (per spec), snapshot it and then be on our merry way.